### ISSUE TITLE
stackedviolin not taking 'order' specification

### ISSUE THREAD

**Issue by GabyBG** [scverse/scanpy#2611]
### Please make sure these conditions are met

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of scanpy.
- [ ] (optional) I have confirmed this bug exists on the master branch of scanpy.

### What happened?

Hello, 

Thank you so much for the beautiful tool that is Scanpy. I am trying to generate stacked violin plots with a particular order for the groups, but the option 'order' does not seem to work. 
I find this issue really weird because categories_order works just fine when I am generating a dotplot.
Maybe I am missing something fundamental.

### Minimal code sample

```python
##code that does not reorder ( I tried both options 'order' and 'categories_order'
sc.pl.stacked_violin(adata,['GATA3','CD8A','CD4'],groupby='sample_id',cmap='PuRd', order=new_order)

sc.pl.stacked_violin(adata,['GATA3','CD8A','CD4'],groupby='sample_id',cmap='PuRd', categories_order=new_order)

##Code that reorders
sc.pl.dotplot(adata,['GATA3','CD8A','CD4'],groupby='sample_id',categories_order=new_order,cmap='PuRd' )
```

### Versions

<details>

```
-----
anndata     0.8.0
scanpy      1.9.3
-----
OpenSSL                     19.0.0
PIL                         10.0.0
apport_python_hook          NA
backcall                    0.2.0
certifi                     2019.11.28
cffi                        1.15.0
chardet                     3.0.4
cloudpickle                 2.2.1
colorama                    0.4.3
colorcet                    3.0.1
cryptography                2.8
cycler                      0.10.0
cython_runtime              NA
cytoolz                     0.12.1
dask                        2023.5.0
dateutil                    2.8.2
debugpy                     1.5.1
decorator                   5.1.0
defusedxml                  0.7.1
entrypoints                 0.3
gseapy                      1.0.5
h5py                        3.7.0
idna                        2.8
igraph                      0.10.6
ipykernel                   6.4.1
ipython_genutils            0.2.0
ipywidgets                  7.6.5
jedi                        0.18.0
jinja2                      3.1.2
joblib                      1.2.0
kiwisolver                  1.4.4
leidenalg                   0.10.0
llvmlite                    0.39.1
lz4                         4.3.2
markupsafe                  2.1.3
matplotlib                  3.6.1
matplotlib_inline           NA
more_itertools              NA
mpl_toolkits                NA
natsort                     8.2.0
netifaces                   0.10.4
numba                       0.56.3
numexpr                     2.8.4
numpy                       1.23.5
packaging                   21.3
pandas                      1.5.3
parso                       0.8.2
patsy                       0.5.3
pexpect                     4.6.0
pickleshare                 0.7.5
pkg_resources               NA
plotly                      5.15.0
prompt_toolkit              3.0.20
psutil                      5.9.4
ptyprocess                  0.7.0
pyarrow                     12.0.1
pydev_ipython               NA
pydevconsole                NA
pydevd                      2.6.0
pydevd_concurrency_analyser NA
pydevd_file_utils           NA
pydevd_plugins              NA
pydevd_tracing              NA
pygments                    2.10.0
pyparsing                   2.4.7
pytz                        2022.4
requests                    2.22.0
scipy                       1.10.1
seaborn                     0.12.0
session_info                1.0.0
setuptools                  68.0.0
simplejson                  3.16.0
sitecustomize               NA
six                         1.14.0
sklearn                     1.3.0
socks                       1.7.1
statsmodels                 0.14.0
storemagic                  NA
tblib                       2.0.0
texttable                   1.6.7
threadpoolctl               3.1.0
tlz                         0.12.1
toolz                       0.12.0
tornado                     6.1
traitlets                   5.1.0
typing_extensions           NA
urllib3                     1.25.8
wcwidth                     0.2.5
yaml                        5.3.1
zipp                        NA
zmq                         22.3.0
zope                        NA
-----
IPython             7.28.0
jupyter_client      7.0.6
jupyter_core        4.8.1
notebook            6.4.5
-----
Python 3.8.10 (default, May 26 2023, 14:05:08) [GCC 9.4.0]
Linux-5.15.0-1040-aws-x86_64-with-glibc2.29
-----
Session information updated at 2023-08-11 23:46
sc.pl.stacked_violin(adata,['GATA3','CD8A','CD4'],groupby='sample_id',cmap='PuRd', order=new_order)
sc.pl.stacked_violin(adata,['GATA3','CD8A','CD
```

</details>

**Comment by flying-sheep** [comment]
> Thank you so much for the beautiful tool that is Scanpy.

thank you! I’m happy to hear that!

> I tried both options 'order' and 'categories_order'

Can you please give a fully reproducible example (i.e. where we just need to copy the code and run it in an empty directory and it shows the problem)

I can’t reproduce this on scanpy master without more context, it seems to work fine:

```py
import scanpy as sc

adata = sc.datasets.krumsiek11()
sc.pl.stacked_violin(
    adata,
    adata.var_names,
    groupby='cell_type',
    use_raw=False,
    title='cat order',
    categories_order=['progenitor', 'Mk', 'Ery', 'Neu', 'Mo'],
)
```

![grafik](https://github.com/scverse/scanpy/assets/291575/4c321471-897e-4276-9b20-fb05f932b2be)

**Comment by GabyBG** [comment]
Hello,

thank you for your help. I was able to find the issue, I have >50 categories that were sorted, and I did not want to input them manually into a list (like your example). After sorting I got an object type 'pandas.core.arrays.categorical.Categorical', that dotplot is able to read the order from, but stacked_violin interpreted differently, using the 'Categories' section.

My solution was as simple as converting to list with:
correct_order = new_order.tolist()

Thank you so much for your help
